source/shared/core_stmt.cpp

```diff
@@ -436,16 +438,21 @@ void core_sqlsrv_bind_param( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT param_
                   ( encoding == SQLSRV_ENCODING_SYSTEM || encoding == SQLSRV_ENCODING_UTF8 || 
                     encoding == SQLSRV_ENCODING_BINARY ), "core_sqlsrv_bind_param: invalid encoding" );
 
-    // if the sql type is unknown, then set the default based on the PHP type passed in
-    if( sql_type == SQL_UNKNOWN_TYPE ) {
-		default_sql_type( stmt, param_num, param_z, encoding, sql_type TSRMLS_CC );
+    if ( stmt->conn->ce_option.enabled && ( sql_type == SQL_UNKNOWN_TYPE || column_size == SQLSRV_UNKNOWN_SIZE ) &&
+         Z_TYPE_P( param_z ) != IS_NULL && Z_TYPE_P( param_z ) != IS_FALSE && Z_TYPE_P( param_z) != IS_TRUE ) {
+        ae_get_sql_type_info( stmt, param_num, direction, param_z, encoding, sql_type, column_size, decimal_digits TSRMLS_CC );
     }
```

Comment:
Such a long list of conditions before calling ae_get_sql_type_info! So if param is null or boolean (true or false), it won't invoke ae_get_sql_type_info? Does it matter whether this is input / output / inout parameter?

Reply:
Yes, if param is null, don't have to all SQLDescribedParam. It doesn't matter what direction it is.